packages/block-library/src/button/index.php

```diff
@@ -17,7 +17,20 @@
  * @return string The block content.
  */
 function render_block_core_button( $attributes, $content ) {
-	$p = new WP_HTML_Tag_Processor( $content );
+	/*
+	 * The current Gutenberg plugin supports WordPress 6.4, but the next_token()
+	 *  method does not exist in WordPress 6.4. Therefore, if Gutenberg is used
+	 * as a plugin, use the Gutenberg class that has the `next_token()` method.
+	 *
+	 * TODO: After the Gutenberg plugin drops support for WordPress 6.4, this
+	 * conditional statement will be removed and the core class `WP_HTML_Tag_Processor`
+	 * should be used.
+	 */
+	if ( defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN && class_exists( 'Gutenberg_HTML_Tag_Processor_6_5' ) ) {
+		$p = new Gutenberg_HTML_Tag_Processor_6_5( $content );
+	} else {
+		$p = new WP_HTML_Tag_Processor( $content );
+	}
```

Comment:
I don't think we want to do this here, preferring instead to just rely on the compatibility layer. When this code gets brought into Core the `Gutenberg_` function should be replaced by the `WP_` one. See `WP_Interactivity_API_Directives_Processor` for another example

Reply:
Thanks for the review!

As far as I can tell, the only class that inherits the `Gutenberg_HTML_Tag_Processor_6_5` class is `WP_Interactivity_API_Directives_Processor`.

Does this mean that the `WP_Interactivity_API_Directives_Processor` class should also be used in this part?

Reply:
No I only meant that we should be fine here using

```php
$p = new Gutenberg_HTML_Tag_Processor_6_5( $content );
```

…and skipping the conditional. Switching back to `WP_HTML_Tag_Processor()` should be part of the porting-into-Core process.

Reply:
I updated via 7835c9f. Are these changes appropriate?

Reply:
Yes they are, but I've asked for some additional help from @ellatrix who ran the last release because I'm concerned about the need to add the ignore statement. Maybe something else is new and I'm unaware.

Reply:
What is the question? I'm not really familiar with any of this

Reply:
it's okay @ellatrix - question was answered. mainly I was unaware that we are sending Gutenberg plugin code into Core now, but apparently that's how people are avoiding name clashes